compiler/test/dotty/tools/dotc/CommentPicklingTest.scala

```diff
@@ -0,0 +1,178 @@
+package dotty.tools.dotc
+
+import dotty.tools.dotc.ast.tpd
+import dotty.tools.dotc.core.Comments.CommentsContext
+import dotty.tools.dotc.core.Contexts.Context
+import dotty.tools.dotc.core.Decorators.PreNamedString
+import dotty.tools.dotc.core.Mode
+import dotty.tools.dotc.core.Names.Name
+import dotty.tools.dotc.core.tasty.DottyUnpickler
+import dotty.tools.dotc.interfaces.Diagnostic.ERROR
+import dotty.tools.dotc.reporting.TestReporter
+
+import dotty.tools.vulpix.{TestConfiguration, ParallelTesting}
+
+import java.io.IOException
+import java.nio.file.{FileSystems, FileVisitOption, FileVisitResult, FileVisitor, Files, Path}
+import java.nio.file.attribute.BasicFileAttributes
+import java.util.EnumSet
+
+import scala.concurrent.duration.{Duration, DurationInt}
+
+import org.junit.Test
+import org.junit.Assert.{assertEquals, assertFalse, fail}
+
+class CommentPicklingTest extends ParallelTesting {
+
+  override def isInteractive: Boolean = false
+  override def testFilter: Option[String] = None
+  override def maxDuration: Duration = 30.seconds
+  override def numberOfSlaves: Int = 5
+  override def safeMode: Boolean = false
+
+  val compileOptions = TestConfiguration.defaultOptions and "-Ykeep-comments" and "-Yemit-tasty"
+  val unpickleOptions = TestConfiguration.defaultOptions
+
+  @Test def commentOnDef: Unit = {
+    val sources = "object A { /** foo */ def bar = 2 }" :: Nil
+    compileAndCheckComment(sources, "bar".toTermName, Some("/** foo */"))
+  }
+
+  @Test def commentOnVal: Unit = {
+    val sources = "object A { /** foo */ val bar = 2 }" :: Nil
+    compileAndCheckComment(sources, "bar".toTermName, Some("/** foo */"))
+  }
+
+  @Test def commentLocalVal: Unit = {
+    val sources = "object A { def buzz = { /** foo */ val bar = 3 } }" :: Nil
+    compileAndCheckComment(sources, "bar".toTermName, Some("/** foo */"))
+  }
+
+  @Test def commentLocalDef: Unit = {
+    val sources = "object A { def buzz = { /** foo */ def bar = 5 } }" :: Nil
+    compileAndCheckComment(sources, "bar".toTermName, Some("/** foo */"))
+  }
+
+  @Test def commentOnClass: Unit = {
+    val sources = "/** foo */ class A" :: Nil
+    compileAndCheckComment(sources, "A".toTypeName, Some("/** foo */"))
+  }
+
+  @Test def commentOnObject: Unit = {
+    val sources = "/** foo */ object A" :: Nil
+    compileAndCheckComment(sources, "A".toTermName, Some("/** foo */"))
+  }
+
+  @Test def commentOnlazyVal: Unit = {
+    val sources = "class A { /** foo */ lazy val buzz = 2 }" :: Nil
+    compileAndCheckComment(sources, "buzz".toTermName, Some("/** foo */"))
+  }
+
+  private def compileAndCheckComment(sources: Seq[String], treeName: Name, expectedComment: Option[String]): Unit = {
```

Comment:
I would use `List` instead of `Seq` across the file. I don't think you gain anything by using `Seq`

Reply:
Do we optimise the API for the most common use case where you want to check for the presence of a comment? `Option[String]` => `String`.

Also maybe add an overload for one source